src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -1030,18 +1026,57 @@ internal bool TryCalculateSyntaxOffsetOfPositionInInitializer(int position, Synt
             // |<--preceding init len-->|      ^
             //                             position
 
-            int initializersLength = isStatic ? membersAndInitializers.StaticInitializersSyntaxLength : membersAndInitializers.InstanceInitializersSyntaxLength;
+            int initializersLength = getInitializersLength(isStatic ? membersAndInitializers.StaticInitializers : membersAndInitializers.InstanceInitializers);
             int distanceFromInitializerStart = position - siblingInitializers[index].Syntax.Span.Start;
 
             int distanceFromCtorBody =
                 initializersLength + ctorInitializerLength -
-                (siblingInitializers[index].PrecedingInitializersLength + distanceFromInitializerStart);
+                (getPrecedingInitializersLength(siblingInitializers, index) + distanceFromInitializerStart);
```

Comment:
>getPrecedingInitializersLength [](start = 17, length = 30)

This doesn't look right because calculation is not going to include lengths from previous groups. I would remove the ```GetInitializersInSourceTree``` function altogether. Located the group and index for the syntax tree and position, then calculated the full length and preceding length in one pass through the entire set of initializers.